src/Engine/GameEvents/CreateUnitEvent.java

```diff
@@ -60,7 +71,29 @@ public void performEvent(MapMaster gameMap)
     if( null != myNewUnit )
     {
       myCommander.units.add(myNewUnit);
-      gameMap.addNewUnit(myNewUnit, myBuildCoords.xCoord, myBuildCoords.yCoord, myBoots);
+      if( myFudgeRadius < 1 || myBoots )
+        gameMap.addNewUnit(myNewUnit, myBuildCoords.xCoord, myBuildCoords.yCoord, myBoots);
+      else
+      {
+        boolean success = false;
+        UnitContext uc = new UnitContext(gameMap, myNewUnit);
+        uc.calculateMoveType();
+        // Check concentric rings outwards, to spawn as close to the original spot as possible
+        for( int radius = 0; !success && radius <= myFudgeRadius; ++radius )
+          for( XYCoord xyc : Utils.findLocationsInRange(gameMap, myBuildCoords, radius, radius) )
+          {
+            Unit resident = gameMap.getResident(xyc);
+            if( resident == null && uc.moveType.canTraverse(gameMap.getEnvironment(xyc)) )
+            {
+              gameMap.addNewUnit(myNewUnit, xyc.xCoord, xyc.yCoord, false);
+              success = true;
+              break;
+            }
+          }
```

Comment:
Don't we typically create Events for effects that have already been determined? E.g., TechDrop finds locations, generates the event, and we can know where they are gonna land. This is figuring out where to create a unit in the Event's `performEvent` call itself. It seems like that logic should be pre-event-creation.

Reply:
What I'm using it for is the resurrection thing - If your hovercraft or fancy fighter dies, it leaves the board entirely and shows up "near" its original location.
At the time that event is created (via listener), the unit will be dead... but because Kaiju step on things to kill them, it's not guaranteed the space you were on before is available.

Reply:
If you are creating a respawn event due to a kaiju movement, wouldn't you know where the kaiju ends up, and be able to find a suitable landing pad for the crushee at event-creation time?

Reply:
If I was creating the respawn event inside of the movement logic... yeah, probably. It'd be a pain because edgecases, but yeah.
However, the resurrection logic is done by a listener for `UnitDieEvent` injecting the new unit creation into the event stream. I don't really want to force any code that might kill a unit to check for whether it resurrects.